src/mail/easey-content-template.service.ts

```diff
@@ -1,44 +1,149 @@
+      this.logger.debug(`Registering partials for ${templateType}`);
+      
+      for (const partialName of config.partials) {
+        try {
+          const partialPath = `${config.basePath}/${partialName}.hbs`;
+          const partialContent = await this.getTemplateContent(partialPath);
+          this.handlebars.registerPartial(partialName, partialContent);
+          this.logger.debug(`Registered partial: ${partialName}`);
+        } catch (error) {
+          this.logger.error(`Failed to register partial ${partialName} from ${config.basePath}`, error);
+        }
+      }
+    }
+  }
 
   returnManager() {
     return this.entityManager;
   }
 
-  async getAndFormatTemplate(templateUrl, context): Promise<string> {
+  // Handlebars template methods
+  async getTemplateContent(templateLocation: string): Promise<string> {
+    const contentUri = this.configService.get<string>('app.contentUri');
+    try {
+      // Handle trailing/leading slashes properly
+      const url = new URL(templateLocation, contentUri).toString();
+      const template = await firstValueFrom(this.httpService.get(url));
+      return template.data;
+    } catch (e) {
+      this.logger.error(`Failed to fetch template: ${templateLocation}`, e);
+      throw new EaseyException(e, HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  async getTemplateById(templateId: number): Promise<EmailTemplate> {
+    const template = await this.entityManager.findOneBy(EmailTemplate, {
+      templateIdentifier: templateId,
+    });
+    if (!template) {
+      throw new EaseyException(
+        new Error(`Template with ID ${templateId} not found`),
+        HttpStatus.NOT_FOUND,
+      );
+    }
+    return template;
+  }
+
+  //This is for rending standard handlebars syntax
+  async renderHandlebarsTemplate(templateLocation: string, context: any): Promise<string> {
+    try {
+      // Ensure partials are registered before rendering (lazy loading)
+      if (!this.partialsRegistered) {
+        await this.registerPartials();
+        this.partialsRegistered = true;
+      }
+
+      const templateString = await this.getTemplateContent(templateLocation);
+      // Compile template (following TemplateService pattern with strict mode)
+      const template = this.handlebars.compile(templateString, { strict: true });
+      return template(context);
+    } catch (error) {
+      this.logger.error(`Failed to render Handlebars template ${templateLocation}`, error);
+      throw error;
+    }
+  }
+
+  //This is for rending simple custom template syntax that is currently using [[]] format.
+  async renderCustomTemplate(templateUrl, context): Promise<string> {
     let templateString;
     const contentUri = this.configService.get<string>('app.contentUri');
     try {
-      const url = `${contentUri}/${templateUrl}`;
+      // Handle trailing/leading slashes properly
+      const url = new URL(templateUrl, contentUri).toString();
```

Comment:
As noted above, this should ensure that the value of `contentUri` ends with a trailing slash so that subdirectories are properly included:
```suggestion
      const url = new URL(templateUrl, contentUri.replace(/\/?$/, '/')).toString();
```

Reply:
Yep! I have updated and pushed this change.